arangod/Cluster/ClusterFeature.cpp

```diff
@@ -560,11 +560,40 @@ void ClusterFeature::start() {
   comm.increment("Current/Version");
 
   ServerState::instance()->setState(ServerState::STATE_SERVING);
+
+  // If we are on a coordinator, we want to have a callback which is called
+  // whenever a hotbackup restore is done:
+  if (role == ServerState::ROLE_COORDINATOR) {
+    auto hotBackupRestoreDone = [this](VPackSlice const& result) -> bool {
+      if (!server().isStopping()) {
+        LOG_TOPIC("12636", INFO, Logger::BACKUP) << "Got a hotbackup restore "
+          "event, getting new cluster-wide unique IDs...";
+        this->_clusterInfo->uniqid(1000000);
+      }
+      return true;
+    };
+    _hotbackupRestoreCallback =
+      std::make_shared<AgencyCallback>(
+        server(), "Sync/HotBackupRestoreDone", hotBackupRestoreDone, true, false);
+    auto r =_agencyCallbackRegistry->registerCallback(_hotbackupRestoreCallback);
+    if (!r) {
+      LOG_TOPIC("82516", WARN, Logger::BACKUP)
+        << "Could not register hotbackup restore callback, this could lead "
+           "to problems after a restore!";
+    }
+  }
 }
 
 void ClusterFeature::beginShutdown() { ClusterComm::instance()->disable(); }
 
 void ClusterFeature::stop() {
+  #ifdef USE_ENTERPRISE
+  if (_hotbackupRestoreCallback != nullptr && !_agencyCallbackRegistry->unregisterCallback(_hotbackupRestoreCallback)) {
```

Comment:
We need to add a check if this is a coordinator here as well, otherwise dbservers and agents issue a strange info message. This was forgotten in the `devel` PR, too.

Reply:
Done.